### PR TITLE
Improve test suite experience

### PR DESCRIPTION
**Pull request by ggravlingen**
#### Reason for change
When adding type hints for the FileSource-file, I'm trying to trace and debug errors. Currently, I'm unable to leverage the benefits of my IDE (VS Code) and its ability to autodiscover all tests and run test with a debugger. The reason for this is that some tests fail by raising different exceptions.

#### Description of change
- Cleanup `get_test_data` by adding type hints, sorting imports and making the code more readable in general.
- Warn instead of throwing exceptions in `get_test_data`. This, together with the changes below, allows us to collect all tests, which enables the test debugger in VS-code.
- Add a check for the zip-file in `test_ixbrl_1_1_conformance_suite.py`. Not doing this causes pytest test collection to fail.
- Ignore `test_esef_ixbrl_conformance_suite.py` when collecting tests. Without this, collecting tests is a very slow process.

#### Steps to Test
I've gotten the sense you guys aren't using VS code but you can try running `pytest --collect-only` before and after this change.

**review**:
@Arelle/arelle

### CODE REVIEW

**Comment by aviary3-wk** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

**Comment by ggravlingen** [comment]
Hmm, this didn't work after the merge of master into this branch. I'll look into why.

File: tests/integration_tests/validation/validation_util.py [hunk 1]
```diff
@@ -1,78 +1,119 @@
...
-                referenced_documents = [case for doc in referenced_documents for case in doc.referencesDocument.keys()]
+                referenced_documents = [
+                    case
+                    for doc in referenced_documents
+                    for case in doc.referencesDocument.keys()
+                ]
             test_cases = sorted(referenced_documents, key=lambda doc: doc.uri)
         else:
-            raise Exception('Unhandled model document type: {}'.format(model_document.type))
+            raise Exception(
```
Comment (sagesmith-wf) [inline tests/integration_tests/validation/validation_util.py:56]: Suggest single line but switch to `f'{}'` syntax instead

File: tests/integration_tests/validation/validation_util.py [hunk 2]
```diff
@@ -1,78 +1,119 @@
...
+                "Unhandled model document type: {}".format(model_document.type)
+            )
         for test_case in test_cases:
             path = PurePosixPath(test_case.uri)
-            test_case_path_tail = path.parts[-3:-1] if path.name == 'index.xml' else path.parts[-2:]
-            if test_case.type not in (ModelDocument.Type.TESTCASE, ModelDocument.Type.REGISTRYTESTCASE):
+            test_case_path_tail = (
```
Comment (sagesmith-wf) [inline tests/integration_tests/validation/validation_util.py:61]: pre-existing: maybe just use pathlib instead?

```python
file_extension = pathlib.Path(test_case.uri).suffix
```

File: tests/integration_tests/validation/validation_util.py [hunk 3]
```diff
@@ -1,78 +1,119 @@
...
+            )
+            if test_case.type not in (
+                ModelDocument.Type.TESTCASE,
+                ModelDocument.Type.REGISTRYTESTCASE,
+            ):
                 test_cases_with_unrecognized_type[test_case_path_tail] = test_case.type
-            if not getattr(test_case, "testcaseVariations", None) and os.path.relpath(test_case.filepath, model_document.modelXbrl.fileSource.basefile) not in expected_empty_testcases:
+            if (
+                not getattr(test_case, "testcaseVariations", None)
+                and os.path.relpath(
```
Comment (sagesmith-wf) [inline tests/integration_tests/validation/validation_util.py:71]: Suggest a local variable for the result here instead of breaking up like this

**Review by sagesmith-wf — APPROVED** [review]

File: tests/integration_tests/validation/validation_util.py [hunk 4]
```diff
@@ -1,78 +1,119 @@
...
             else:
                 for mv in test_case.testcaseVariations:
-                    test_id = '{}/{}'.format('/'.join(test_case_path_tail), mv.id)
+                    test_id = "{}/{}".format("/".join(test_case_path_tail), mv.id)
                     param = pytest.param(
                         {
-                            'status': mv.status,
-                            'expected': mv.expected,
-                            'actual': mv.actual
+                            "status": mv.status,
```
[the remark this answers is not in the document]
Comment (ggravlingen) [reply tests/integration_tests/validation/validation_util.py:82]: I'll remove status instead, it adds no benefits here.

File: conftest.py [hunk 5]
```diff
@@ -21,3 +27,8 @@ def pytest_addoption(parser):
     parser.addoption('--tests', default=tests_default,
                      help='.ini file to load test suites from (default is arelle/confi/arelle_test.ini)')
 
+
+collect_ignore = [
+    # Test collection of this file is very slow so we ignore it
+    "tests/integration_tests/validation/ESEF/test_esef_ixbrl_conformance_suite.py"
```
[the remark this answers is not in the document]
Comment (ggravlingen) [reply conftest.py:33]: This is only exercised when running `pytest tests`. Normally, this file is executed through some other method in the CI/CD?

File: tests/integration_tests/validation/validation_util.py [hunk 6]
```diff
@@ -1,78 +1,119 @@
...
+            ModelDocument.Type.TESTCASE,
+            ModelDocument.Type.REGISTRYTESTCASE,
+        ):
             test_cases.append(model_document)
         elif model_document.type == ModelDocument.Type.TESTCASESINDEX:
             referenced_documents = model_document.referencesDocument.keys()
             child_document_types = {doc.type for doc in referenced_documents}
-            assert len(child_document_types) == 1, f'Multiple child document types found in {model_document.uri}: {child_document_types}.'
+            assert (
+                len(child_document_types) == 1
```
[the remark this answers is not in the document]
Comment (ggravlingen) [reply tests/integration_tests/validation/validation_util.py:44]: I found the file very hard to read so I ran Black on it but I'm happy to revert this line.

File: tests/integration_tests/validation/validation_util.py [hunk 7]
```diff
@@ -1,78 +1,119 @@
...
-        nonexistent_expected_failure_ids = expected_failure_ids - set(test_id_frequencies)
+            warnings.warn(
+                f"Some test IDs are not unique.  Frequencies of nonunique test IDs: {nonunique_test_ids}."
+            )
+        nonexistent_expected_failure_ids = expected_failure_ids - set(
+            test_id_frequencies
+        )
         if nonexistent_expected_failure_ids:
-            raise Exception(f"Some expected failure IDs don't match any test cases: {sorted(nonexistent_expected_failure_ids)}.")
+            warnings.warn(
```
[the remark this answers is not in the document]
Comment (ggravlingen) [reply tests/integration_tests/validation/validation_util.py:110]: Autodiscovery runs on all tests and these lines threw exceptions causing autodiscovery not to work.

**Comment by ggravlingen** [comment]
Considering your comments here, I think it's best to close this and see if I can work around the problem extending my `python.testing.pytestArgs` config in the VS code config. Thanks for reviewing this.

**Pull request closed without merging**